arduino-ide-extension/src/node/core-service-impl.ts

```diff
@@ -26,6 +26,7 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
         const { sketchUri, fqbn, compilerWarnings } = options;
         const sketchPath = FileUri.fsPath(sketchUri);
 
+        await this.coreClientProvider.initialized;
```

Comment:
@silvanocerza what happens if the `initialized` fails or rejects? is it ok to proceed calling `coreClient()`?

Reply:
I think it's fine, if the `Init` gRPC function fails it means it has been called, so `Create` was successful meaning we have an internal instance in the CLI daemon and we have a "working" client.

If `Init` fails completely though we must have some other huge problems, with the refactoring failure to load a platform or library just logs an error, but it doesn't stop completely.

I still don't like how I implemented this though, I'd like to show the user a notification when there are this kind of failures but am not sure how to handle that.